lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -85,5 +85,13 @@ def sanitize_shop_param(params)
       return unless params[:shop].present?
       ShopifyApp::Utils.sanitize_shop_domain(params[:shop])
     end
+
+    def sanitized_params
+      query_params = request.query_parameters
+      return query_params unless params[:shop].is_a?(String)
+
+      query_params[:shop] = sanitize_shop_param(params)
+      query_params
```

Comment:
The code in this method is fairly procedural. I think we can make it more OO and concise. How about something like this?

```ruby
request.query_parameters.clone.tap do |query_params|
  query_params[:shop] = sanitize_shop_params(params) if params[:shop].is_a?(String)
end
```

Reply:
Can we also write a test for the `params[:shop].is_a?(String)` check. That'll serve as documentation for when someone tries to remove this.

Reply:
Thanks very much @nickhoffman - `tap` is definitely a much nicer approach.

I've also added a couple tests for the `login_url` and `return_to` URL's containing a non-String shop param